File: financeTracker/src/main/java/com/springapp/financetracker/service/income/IncomeService.java
package com.springapp.financetracker.service.income;

import com.springapp.financetracker.controller.payload.NewIncomePayload;
import com.springapp.financetracker.controller.payload.UpdateIncomePayload;
import com.springapp.financetracker.dto.IncomeResponseDto;
import com.springapp.financetracker.entity.Income;

import java.util.List;

public interface IncomeService {
    IncomeResponseDto addIncome(NewIncomePayload payload);
    IncomeResponseDto getIncome(Integer id);
    List<IncomeResponseDto> getAllIncome();
    IncomeResponseDto updateIncome(UpdateIncomePayload payload);
    void deleteIncome(Integer incomeId);
}

File: frontend/src/services/incomeService.ts
import { api } from "./api"

export const getAllIncome = async() => {
    const response = await api.get("/income/all");
    return response.data;
}

export const addIncome = async(data:any) =>{
    const response = await api.post("/income/add", data);
    return response.data;
}

export const getIncome = async(incomeId:number) =>{
    const response = await api.get(`/income/${incomeId}`);
    return response.data;
}

export const updateIncome = async(data:any) =>{
    const response = await api.post(`/income/update/${data.incomeId}`, data);
    return response.data;
}

export const deleteIncome = async(incomeId:number) => {
    await api.post(`/income/delete/${incomeId}`);
}